Bearded.Graphics/Debugging/KHRDebugExtension.cs

```diff
@@ -0,0 +1,57 @@
+using System.Linq;
+using OpenTK.Graphics.OpenGL;
+
+namespace Bearded.Graphics.Debugging
+{
+    public abstract class KHRDebugExtension
+    {
+        private static KHRDebugExtension? instance;
+        public static KHRDebugExtension Instance => instance ??= createInstanceWithAvailableFunctionality();
+
+        private KHRDebugExtension()
+        {
+        }
+
+        private static KHRDebugExtension createInstanceWithAvailableFunctionality()
+        {
+            if (khrDebugExtensionIsAvailable())
```

Comment:
nit: I find this stylistically a strange choice between compactness (not using brackets) but also not choosing to use a ternary expression. Is there a reason we're not reducing this to `khrDebugExtensionIsAvailable() ? new ActualImplementation() : new DummyImplementation()`?

Reply:
Your code would not compile because the two aren't the same type and it can't figure out what base type to use without a cast, and that's even uglier than what it is right now I think.